rust/src/connection/network/channel.rs

```diff
@@ -49,8 +52,13 @@ impl GRPCChannel for PlainTextChannel {}
 
 impl GRPCChannel for CallCredChannel {}
 
+const TIMEOUT: Duration = Duration::from_secs(60);
```

Comment:
Chosen arbitrarily. Anything less than a second or so is too strict for slow connections, but going significantly over a minute would probably be too long.

Reply:
The query does time out if the server does not respond within this time (tested by adding a `sleep(60000)` in `TransactionService::respond()` on the server side).

The default transaction timeout is 5 minutes. I'd extend this timeout to 10 minutes or even an hour.

Ideally we'd have a connection builder that would let the user specify the desired request timeout, but that would take a lot longer.